src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1665,6 +1740,32 @@ UndistributeTable(Oid relationId)
 							NULL, None_Receiver, NULL);
 	}
 
+	if (conversionType == ALTER_DISTRIBUTED_TABLE)
+	{
+		int tempGlobalShardCount = ShardCount;
+		if (shardCount)
+		{
+			ShardCount = shardCount;
```

Comment:
Would it be possible to refactor CreateDistributedTable to have a shard count parameter and pass `ShardCount` in other call sites?